docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,276 @@
+
+1. Get your Hugging Face API token:
+
+   ```python
+   >>> from huggingface_hub import HfFolder
+   >>> token = HfFolder().get_token()
+   ```
+2. Use `create_repo` to create your repository:
+
+   ```python
+   >>> from huggingface_hub import HfApi
+   >>> api = HfApi()
+   >>> api.create_repo(token=token, name="test-model")
+   'https://huggingface.co/lysandre/test-model'
+   ```
+
+### Delete a repository
+
+Delete a repository with `delete_repo`. Make sure you are certain you want to delete a repository because this is an irreversible process!
+
+Pass your token and the name of the repository to `delete_repo`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.delete_repo(token=token, name=REPO_NAME)
+```
+
+Delete a dataset repository by adding the `repo_type` parameter:
+
+```python
+>>> api.delete_repo(token=token, name=REPO_NAME, repo_type="dataset")
+```
+
+### Change repository visibility
+
+A repository can be public or private. A private repository is only visible to you or members of your organization. Change a repository to private as shown in the following:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.update_repo_visibility(token=token, name=REPO_NAME, private=True)
+```
+
+## `Repository` 
+
+The `Repository` class allows you to push models or other repositories to the Hub. `Repository` is a wrapper over Git and Git-LFS methods, so make sure you have Git-LFS installed (see [here](https://git-lfs.github.com/) for installation instructions) and set up before you begin. The `Repository` class should feel familiar if you are already familiar with common Git commands. 
+
+### Clone a repository
+
+The `clone_from` parameter clones a repository from a Hugging Face model ID to a directory specified by the `local_dir` argument:
+
+```python
+>>> repo = Repository(local_dir="w2v2", clone_from="facebook/wav2vec2-large-960h-lv60")
+```
+
+`clone_from` can also clone a repository from a specified directory using a URL (if you are working offline, this parameter should be `None`):
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="https://github.com/huggingface/huggingface_hub")
+```
+
+Easily combine the `clone_from` parameter with `create_repo` to create and clone a repository:
+
+```python
+>>> repo_url = HfApi().create_repo(name="repo_name")
+>>> repo = Repository(local_dir="repo_local_path", clone_from=repo_url)
+```
+
+### Using a local clone
+
+Instantiate a `Repository` object with a path to a local Git clone or repository:
+
+```python
+>>> from huggingface_hub import Repository
+>>> repo = Repository(local_dir="<path>/<to>/<folder>")
+```
+
+### Commit and push to a cloned repository
+
+If you want to commit or push to a cloned repository that belongs to you or your organizations:
+
+1. Log in to your Hugging Face account with the following command:
+
+   ```bash
+   huggingface-cli login
```

Comment:
Perhaps it would be useful for Jupyter notebook users to know about `notebook_login()` here? A common pattern would be to try running:

```
!huggingface-cli login
```

but that won't work in Jupyter notebooks. You can display the widget as follows:

```python
from huggingface_hub import notebook_login

notebook_login()
```

Reply:
Wow, that's really cool! I didn't know this was possible. Added a step for using `notebook_login`.